### PR TITLE
Support comparison mode in mobile

### LINKED ISSUE #3188: Allow comparison mode to work on mobile
Comparison mode partially works on mobile, if you go to a comparison link like: https://go.nasa.gov/3ldGVsu, swipe appears and you can compare between the images. When you open the layer list, it shows the "B" side layer list, and when you close the layer list, you are no longer in compare mode, and will be viewing the "B" date. 

We should allow Comparison mode to work on mobile and to display the two layer lists.

### PR DESCRIPTION
**Pull request by edplato**
## Description

Fixes #3188  .

- [x] Allow "official support" to comparison mode in mobile. Comparison mode will be limited to `swipe` in mobile and will use the same sidebar A/B tabs along with a new A/B toggle button.
- [x] Add initial mobile screen size check to config on load as `config.initialIsMobile` to help determine location state changes for compare permalinks opened in mobile
- [x] Add e2e mobile-compare tests and minor update on a few others.
- [x] Minor cleanup removing unnecessary dom depth, remove unused function params, and random spell fixes from adding a new extension to VS Code `Code Spell Checker`

## How To Test

1. Activate compare mode in mobile, verify activate/deactivate, verify desktop permalinks with compare mode open in mobile including a desktop `spy` compare permalink should open as a `swipe` when opened on a mobile device

## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/master/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview

### CODE REVIEW

File: web/js/containers/sidebar/layers-container.js [hunk 1]
```diff
@@ -103,8 +105,9 @@ function LayersContainer (props) {
     </DragDropContext>
   );
 
+  const maxHeight = isCompareActive && isMobile ? height - 94 : height;
```
Comment (jasontk19) [inline web/js/containers/sidebar/layers-container.js:108]: It looks like the compare button gets pushed off screen when there are lots of layers in the layer list:
Screen Shot 2021-03-17 at 10 03 03 AM

**Review by jasontk19 — CHANGES_REQUESTED** [review]
This looks great!  Tried it out on my iPhone and other than the button positioning issue it works really well and feels like it was intended for mobile use.  I uploaded it to GIBS UAT for on-device testing here: https://uat.gibs.earthdata.nasa.gov/worldview-dev/mobile-compare/

**Review by Benjaki2 — CHANGES_REQUESTED** [review]
Looks really good!! I found one minor issue... it looks like the permalink is now updating on load before any changes are made.

File: web/js/containers/timeline/timeline.js [hunk 2]
```diff
@@ -965,95 +967,116 @@ class Timeline extends React.Component {
...
+  * @returns {Object} style { left, bottom }
+  */
+  getMobileDateButtonStyle = () => {
+    const {
+      hasSubdailyLayers,
+      isCompareModeActive,
+      isScreenWidthLessThan350,
+      isScreenWidthLessThan450,
+    } = this.props;
+    // eslint-disable-next-line no-nested-ternary
```
[the remark this answers is not in the document]
Comment (edplato) [reply web/js/containers/timeline/timeline.js:992]: I do like simple, readable nested ternaries, but there is a lot going on in this one, so I'll try to clean it up.

File: web/js/main.js [hunk 3]
```diff
@@ -144,6 +144,9 @@ window.onload = () => {
         parameters.mockTour = isMockTour;
       }
 
+      const crs = calculateResponsiveState(window);
+      config.initialIsMobile = crs.innerWidth <= 768;
```
[the remark this answers is not in the document]
Comment (edplato) [reply web/js/main.js:148]: My main motivation for this is to determine the browser size as mobile during `location.js` parsing for conditional checks/revisions of `spy`/`opacity` modes in mobile. I will look into accessing the `state.browser` in the `getParameters()` `location.js` somehow.

[the remark this answers is not in the document]
Comment (jasontk19) [reply web/js/main.js:148]: Ah, right.  I didn't look at this closely enough to realize this was happening before the store gets initialized.  If you can find an easy way to do this, go for it.  But I wouldn't worry about refactoring too much just to get this one property in a different place on state.

File: web/js/containers/timeline/timeline.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jasontk19) [reply web/js/containers/timeline/timeline.js:992]: Sounds like we have more votes for keeping them so, I would say go ahead and remove the eslint rule so we can stop disabling it each time we use them.

**Comment by edplato** [comment]
The overflow product picker buttons issue should be fixed. Along with that I removed the nested-ternary lint warning, cleaned up the timeline mobile datepicker nested ternary logic, and removed the timeline `display: none` condition to now just not render it as suggested.

I also fixed this subtle issue where the `footer .product-buttons` weren't quite contained that I found in the process of investigating sidebar styling:
image

**Comment by edplato** [comment]
> Looks really good!! I found one minor issue... it looks like the permalink is now updating on load before any changes are made.

@Benjaki2 Could you expand on this issue. I can't narrow down exactly what is updating.

**Comment by Benjaki2** [comment]
> > Looks really good!! I found one minor issue... it looks like the permalink is now updating on load before any changes are made.
> 
> @Benjaki2 Could you expand on this issue. I can't narrow down exactly what is updating.

This is what I am seeing `onload`
Screen Shot 2021-03-19 at 10 55 54 AM

**Comment by edplato** [comment]
Ok, I'm seeing what you're seeing too. It looks like this in PROD also. This appears to be a change where localStorage is set to `groupOverlays: disabled` and then on page load the `lg` query string is added even if it's an empty query string default page. Then the `t` query string is getting added from a recent change where anytime there's any query string params it's set up to add the time too (mainly to fix users sharing links with no times from copy/pasting the url instead of going through our share link tool).

**Comment by Benjaki2** [comment]
@edplato That makes sense. I was comparing this to prod, but I must not have that applied in the localStorage there. 💯

**Review by Benjaki2 — APPROVED** [review]

**Review by jasontk19 — APPROVED** [review]
Looks good to me!

**Pull request merged**